src/index.ts

```diff
@@ -150,10 +153,17 @@ export default function createIntegration(args?: Options): AstroIntegration {
           if (!file.endsWith(".mjs")) continue;
           const pth = fileURLToPath(new URL(file, chunksDirUrl));
           const contents = fs.readFileSync(pth, "utf-8");
+          if (
+            !contents.includes(DENO_IMPORTS_SHIM_LEGACY) &&
+            !contents.includes(DENO_IMPORTS_SHIM)
+          ) continue;
```

Comment:
This way we avoid running `writeFileSync` on files that don't need it.

Reply:
Nicely done!